core/src/main/java/bisq/core/dao/governance/blindvote/BlindVoteListService.java

```diff
@@ -142,18 +157,30 @@ private void onAppendOnlyDataAdded(PersistableNetworkPayload persistableNetworkP
             if (!blindVotePayloads.contains(blindVotePayload)) {
                 BlindVote blindVote = blindVotePayload.getBlindVote();
                 String txId = blindVote.getTxId();
-                // We don't validate as we might receive blindVotes from other cycles or phases at startup.
-                // Beside that we might receive payloads we requested at the vote result phase in case we missed some
-                // payloads. We prefer here resilience over protection against late publishing attacks.
+
                 if (blindVoteValidator.areDataFieldsValid(blindVote)) {
                     if (fromBroadcastMessage) {
-                        log.info("We received a blindVotePayload. blindVoteTxId={}", txId);
+                        if (isInCorrectPhase(daoStateService.getChainHeight())) {
+                            // We received the payload outside the vote reveal phase and add the payload.
+                            // If we would accept it during the vote reveal phase we would be vulnerable to a late
+                            // publishing attack where the attacker tries to pollute the data view of the voters and
+                            // render the whole voting cycle invalid if the majority hash is not at least 80% of the
+                            // vote stake.
+                            blindVotePayloads.add(blindVotePayload);
+                        }
+                    } else {
+                        // In case we received the data from the seed node at startup we cannot apply the phase check as
+                        // even in the vote reveal phase we want to receive missed blind votes.
+                        blindVotePayloads.add(blindVotePayload);
                     }
-                    blindVotePayloads.add(blindVotePayload);
                 } else {
                     log.warn("We received an invalid blindVotePayload. blindVoteTxId={}", txId);
                 }
             }
         }
     }
+
+    private boolean isInCorrectPhase(int blockHeight) {
```

Comment:
It would be better to just call this `notInRevealPhase` or `isAcceptingProposals` to better describe what it does.

Reply:
Ok, will rename (had it that way before ;-))